libs/community/langchain_community/vectorstores/sqlitecloud_vec.py

```diff
@@ -0,0 +1,205 @@
+from __future__ import annotations
+
+import enum
+import json
+import logging
+import warnings
+from typing import (
+    TYPE_CHECKING,
+    Any,
+    Iterable,
+    List,
+    Optional,
+    Tuple,
+    Type,
+)
+
+from langchain_core.documents import Document
+from langchain_core.embeddings import Embeddings
+from langchain_core.vectorstores import VectorStore
+
+if TYPE_CHECKING:
+    import sqlitecloud
+
+logger = logging.getLogger(__name__)
+
+_LANGCHAIN_DEFAULT_TABLE_NAME = "langchain_store"
+
+class SQLiteCloudVec(VectorStore):
+    """SQLite Cloud as a vector database.
+
+    To use, you should have a SQLite Cloud account and python packages installed.
+    Example:
+        .. code-block:: python
+            from langchain_community.vectorstores import SQLiteCloudVec
+            from langchain_community.embeddings.openai import OpenAIEmbeddings
+            ...
+    """
+    class VectorType(str, enum.Enum):
+        """Enumerator for the supported Vector types"""
+        FLOAT = "float"
+        BIT = "bit"
+
+    def __init__(
+        self,
+        embedding: Embeddings,
+        conn_str: str,
+        db_name: str,
+        table_name: str = _LANGCHAIN_DEFAULT_TABLE_NAME, # type: ignore
+        logger: Optional[logging.Logger] = logger,
+    ):  
+        base_url, params = conn_str.split('?', 1)
+
+        # Insert the database name between the base URL and query parameters
+        formatted_conn_str = f"{base_url}/{db_name}?{params}"
+
+        connection = sqlitecloud.connect(formatted_conn_str)
+
+        if not isinstance(embedding, Embeddings):
+            warnings.warn("embeddings input must be Embeddings object.")
+
+        self._connection = connection
+        self._table_name = table_name
+        self._embedding = embedding
+        self._logger = logger
+        self._vector_type = self.VectorType.FLOAT
+
+        self.create_table_if_not_exists()
+
+    def create_table_if_not_exists(self) -> None:
+        self._connection.execute(
+            f"""
+                CREATE VIRTUAL TABLE IF NOT EXISTS ? USING vec0(
```

Comment:
better to replace in the string the name of the variable. In sqlite we cannot use qmark to replace table name, just variables

```
f"CREATE VIRTUAL TABLE IF NOT EXISTS {self._table_name} USING vec0(..."
```